search_engine.py

```diff
@@ -0,0 +1,51 @@
+import sqlite3
+from sqlite3 import Error
+
+
+
+def create_connection(db_file):
+    """ create a database connection to the SQLite database
+        specified by the db_file
+    :param db_file: database file
+    :return: Connection object or None
+    """
+    conn = None
+    try:
+        conn = sqlite3.connect(db_file)
+    except Error as e:
+        print(e)
+
+    return conn
+
+
+def select_preferences_by_UserID(conn, User):
+    """
+    Query preferences by UserID
+    :param conn: the Connection object
+    :param UserID:
+    :return:
+    """
+    cur = conn.cursor()
+    cur.execute("SELECT * FROM Users WHERE User=?", (User,))
```

Comment:
Do you have Users table? while i uploaded the Vagrant files, i did not see there is Users table,
BTW django is not up, please rebase Vagrant file.
If you have new Users Table different than djando user_authentication